src/Analyzers/Core/Analyzers/SimplifyInterpolation/Helpers.cs

```diff
@@ -132,6 +136,59 @@ private static IOperation Unwrap(IOperation expression)
             formatString = null;
         }
 
+        private static bool IsObjectToStringOverride(IMethodSymbol method)
+        {
+            while (method.OverriddenMethod is not null)
+                method = method.OverriddenMethod;
+
+            return method.ContainingType.SpecialType == SpecialType.System_Object
+                && method.Name == nameof(ToString);
+        }
+
+        private static bool IsInvariantCultureReference(IOperation operation)
+        {
+            if (Unwrap(operation) is not IPropertyReferenceOperation { Member: { } member })
+                return false;
+
+            return member.Name switch
+            {
+                nameof(CultureInfo.InvariantCulture) => SymbolEqualityComparer.Default.Equals(
+                    member.ContainingType,
+                    operation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.Globalization.CultureInfo).FullName!)),
+
+                "InvariantInfo" =>
+                    SymbolEqualityComparer.Default.Equals(
+                        member.ContainingType,
+                        operation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.Globalization.NumberFormatInfo).FullName!))
+                    || SymbolEqualityComparer.Default.Equals(
+                        member.ContainingType,
+                        operation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.Globalization.DateTimeFormatInfo).FullName!)),
+
+                _ => false,
+            };
+        }
+
+        private static bool IsInsideFormattableStringInvariant(IOperation operation)
+        {
+            var interpolatedStringOperation = AncestorsAndSelf(operation).OfType<IInterpolatedStringOperation>().FirstOrDefault();
+
+            return Unwrap(interpolatedStringOperation?.Parent, towardsParent: true) is IArgumentOperation
+            {
+                Parent: IInvocationOperation
+                {
+                    TargetMethod: { Name: nameof(FormattableString.Invariant), ContainingType: var containingType },
+                },
+            } && SymbolEqualityComparer.Default.Equals(containingType, operation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.FormattableString).FullName!));
```

Comment:
can you break out helper for the .equals call here?  you use this pattern in a few places.